src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -102,8 +144,23 @@ private void CompilationStartAction(CompilationStartAnalysisContext context)
             var builder = ImmutableDictionary.CreateBuilder<string, string>();
             builder[nameof(NamingStyle)] = applicableRule.NamingStyle.CreateXElement().ToString();
             builder["OptionName"] = nameof(SimplificationOptions.NamingPreferences);
-            builder["OptionLanguage"] = context.Compilation.Language;
-            context.ReportDiagnostic(Diagnostic.Create(descriptor, context.Symbol.Locations.First(), builder.ToImmutable()));
+            builder["OptionLanguage"] = compilation.Language;
+
+            return Diagnostic.Create(descriptor, symbol.Locations.First(), builder.ToImmutable());
+
+            // Local functions
+
+            async Task<NamingStylePreferences> GetNamingStylePreferencesAsync()
+            {
+                var location = symbol.Locations.FirstOrDefault();
+                if (location == null)
+                {
+                    return null;
+                }
+
+                var optionSet = await options.GetDocumentOptionSetAsync(location.SourceTree, cancellationToken).ConfigureAwait(false);
```

Comment:
i'm a littler nervous about this.  can we check that we have a source tree first?

Reply:
sure (even though I can't think of a scenario where this would occur, so I'm not adding a test)

Reply:
fixed in 926a46847bdb00abb502a3088863ebd3fcfa04ca